src/main/scala/viper/server/VerificationWorker.scala

```diff
@@ -14,26 +14,33 @@ import scala.collection.mutable.ListBuffer
 import akka.actor.ActorRef
 import ch.qos.logback.classic.Logger
 import viper.carbon.CarbonFrontend
-import viper.server.ViperServerRunner.ReporterActor
 import viper.silicon.SiliconFrontend
 import viper.silver.ast.{Position, _}
 import viper.silver.frontend.{DefaultStates, SilFrontend}
 import viper.silver.reporter
 import viper.silver.reporter.{Reporter, _}
 import viper.silver.verifier.errors._
 import viper.silver.verifier.{AbstractVerificationError, _}
+import viper.silver.verifier.VerificationResult
 
 import scala.language.postfixOps
 
+
+
 // Implementation of the Reporter interface used by the backend.
-class ActorReporter(private val actor_ref: ActorRef, val tag: String)
+class ActorReporter(private val actor_ref: ActorRef, private val rep: Option[Reporter], val tag: String)
   extends viper.silver.reporter.Reporter {
 
   val name = s"ViperServer_$tag"
 
   def report(msg: reporter.Message): Unit = {
     //println(s"ActorReporter reporting >>> ${msg}")
-    actor_ref ! ReporterActor.ServerReport(msg)
+    actor_ref ! ReporterProtocol.ServerReport(msg)
+
+    rep match {
+      case Some(reporter) => reporter.report(msg)
```

Comment:
This allows the reporting of messages via the reporter which is given as an argument to the verify method of ViperCoreServer.

Reply:
Thanks for the explanation. Is there a use case in Gobra IDE when you need to change the reporter for a particular instance of ViperCoreServer? I'm wondering why we cannot have it fixed.

Reply:
We put our Go error reporting inside of the Viper reporter. This has to be created for each Viper AST.

Reply:
Thanks, but I still don't understand quite fully: ActorReporter is already a reporter. Why does it need to aggregate another reporter? 

If what you want is to map verification results (e.g., from Viper to Gobra) that are reported to an ActorReporter, you could extend ActorReporter in a similar way as we did in [PluginAwareReporter](https://github.com/viperproject/silver/blob/cc632f9907ec91baffc568b2c847c7c86f277aaa/src/main/scala/viper/silver/plugin/PluginAwareReporter.scala). Maybe PluginAwareReported would do exactly what you want, actually, but its name is not ideas... should be called ResultTransformingReporter, or something like that. 

Currently, ActorReporter sends messages to its instance of ReporterActor **and** forwards the same messages to its ```rep```. I find this duplication confusing.

Reply:
I have zero clue what the ActorReporter is and why it should be leaked to the interface of the ViperCoreServer. The backends (silicon and carbon) use - besides their result - also reporters to communicate errors. Consider the line "assert 0 > x.f.val" in a Viper program. If the assertion fails, it will report a AssertionError message with the according error and everything. In Gobra, and any other frontend, we have to take this Viper error and transform it into an error of the frontend. Instead of polling on the results of the Viper reporter, we wrote a wrapper of the Viper reporter that wraps a GobraReporter, but accepts Viper error message, which it translates back to Gobra errors. Currently, we give this wrapped reporter to the backends, when we verify a file with silicon or carbon. 
In our meeting, we expressed the need of continued processing such as the error backtranslation, which should not require any polling. To satisfy this need, we included the reporter as an argument of the verify method in our ViperCoreServer. 

Questions from my side: 1) What is the ActorReporter and why do we have to leak details about it to the ViperCoreServer, 2) How is this PluginAwareReporter provided to ViperServer? In particular, can it be provided for each program ast? If not, then it is not sufficient for us. If it can, then why do we not just use the Viper reporter, which - to my knowledge - does not constrain the image of the error transformation, 3) why can we not just have a reporter as an argument as agreed on in the ViperCoreServer interface, which we discussed all together? Also PluginAwareReporter does not do what we want because it can only transform from Viper error to Viper error which is useless to us.

Reply:
Also I do not understand you last sentence at all. What is ReporterActor? What is the "rep" that messages are forwarded to? If this "rep" is a reporter, then isn't this literally what we want?